tutorials/math/matrix-exponentiation.md

```diff
@@ -9,8 +9,522 @@ keywords:
   - matrix exponentiation
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@Ishwarendra"/>
 
 ## Overview
 
-...
\ No newline at end of file
+First and foremost, it is **highly recommended** to go through [Binary Exponentiation](./number-theory/binary-exponentiation.md), if you are not acquainted with it or don't remember it clearly. It is expected that reader knows Multiplication of two matrices.
+
+Matrix exponentiation is very useful in solving problems that involve finding $n^{th}$ term of a [linear recurrence with constant coefficients](https://en.wikipedia.org/wiki/Linear_recurrence_with_constant_coefficients) in order of $log(n)$. 
+
+Matrix exponentiation is all about finding a matrix $T$ such that $T \cdot A = B$ where $A$ and $B$ are two matrices of appropriate order. Before we delve deeper let's us recall two things:
+1. Matrix Multiplication is **associative** $i.e.$, $A \cdot (B \cdot C) = (A \cdot B) \cdot C$ 
+2. A matrix $A$ of order $(n \times m)$ can be multiplied with another matrix of order $(m \times p)$ **only**. 
+
+$Q.1.$ Before we go any further, let us solve a problem. Find a matrix $T$ such that $T \cdot A$ = $B$, where
+
+$$$
+A = \begin{bmatrix}
+a \\
+b \\
+c
+\end{bmatrix}
+
+;
+
+B = \begin{bmatrix}
+2 a + 3 b \\
+a + 2 b + 4 c
+\end{bmatrix}
+$$$
+
+<details>
+
+<summary> Answer </summary>
+
+$$$
+T = \begin{bmatrix}
+2 & 3 & 0 \\
+1 & 2 & 2
+\end{bmatrix}
+$$$
+
+Verify the following yourself: 
+
+$$$ 
+\begin{bmatrix}
+2 & 3 & 0 \\
+1 & 2 & 2
+\end{bmatrix} \cdot
+\begin{bmatrix}
+a \\
+b \\
+c
+\end{bmatrix}
+= 
+\begin{bmatrix}
+2 a + 3 b \\
+a + 2 b + 4 c
+\end{bmatrix}
+$$$
+
+If you are struggling with matrix multiplication remember that $B_{i, j}$ comes from sum and product of $i^{th}$ **row** of $A$ and $j^{th}$ **column** of $B$.
+
+</details>
+
+$Q.2.$ Find a matrix $T$ such that $T \cdot A = B$, where
+
+$$$A = \begin{bmatrix}
+a \\
+b \\
+c
+\end{bmatrix}
+
+B = \begin{bmatrix}
+a + 3b \\
+2b + 3c \\
+5a + 3b + 7c \\
+6a
+\end{bmatrix}
+$$$
+
+<details>
+
+<summary> Answer </summary>
+
+$$$
+T = \begin{bmatrix}
+1 & 3 & 0 \\
+0 & 2 & 3 \\
+5 & 3 & 7 \\
+6 & 0 & 0
+\end{bmatrix}
+$$$ 
+Verify it yourself!
+
+</details>
+
+Let us now see a concrete problem that can be solved using Matrix Exponentiation.
+
+### [1. Fibonacci Number](https://leetcode.com/problems/fibonacci-number/description/)
```

Comment:
Change the title to `Example #1: 509. Fibonacci Number`

Reply:
Fixed.